pkg/versioninfo/version.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2023 The ClusterLink Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package versioninfo
+
+import (
+	"runtime/debug"
+	"strings"
+	"time"
+)
+
+func init() {
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		return
+	}
+
+	for _, kv := range info.Settings {
+		switch kv.Key {
+		case "vcs.revision":
+			Revision = kv.Value
+		case "vcs.time":
+			LastCommit, _ = time.Parse(time.RFC3339, kv.Value)
+		case "vcs.modified":
+			DirtyBuild = kv.Value == "true"
+		}
+	}
+}
+
+// Short provides a short string summarizing available version information.
+// The format is <SemVer>-<GIT SHA>[-<dirty>]
+func Short() string {
+	parts := make([]string, 0, 4)
+
+	if SemVer != "" {
+		parts = append(parts, SemVer)
+	}
+	if Version != "unknown" && Version != "(devel)" {
```

Comment:
I didn't saw in the code where you update the version

Reply:
Version will be the version tag if the binary is built with "go install url/package@version". If the binary is built some other way, it will be "(devel)".
So in our case you're correct - it'll remain as "devel"/"unknown". I can handle it in a follow up PR based on the TODO in varaibles.go